### ISSUE TITLE
Combining i18n with namespace for error messages

### ISSUE THREAD

**Issue by jmarceli** [dry-rb/dry-validation#322]
It seems that when I combine `i18n` with `namespace` configuration the message root locale goes back to the default `:en`.
Here is my `config/locales/errors.pl.yml` file:
```yaml
pl:
  errors:
    filled?: Pole nie może pozostać puste
    str?: Wartość musi być tekstem

    rules:
      note:
        rules:
          note:
            filled?: Alternatywny komunikat o pustym polu
```

Here is the code:
```ruby
      schema = Dry::Validation.Form do
        configure {
          config.namespace = :note
          config.messages = :i18n
        }
        required(:note).filled(:str?)
      end

      schema.call(params.to_unsafe_h).messages
```

In order to make it work properly I have to set locale explicitly as a messages `locale` argument:
```ruby
      schema = Dry::Validation.Form do
        configure {
          config.namespace = :note
          config.messages = :i18n
        }
        required(:note).filled(:str?)
      end

      schema.call(params.to_unsafe_h).messages(locale: I18n.default_locale)
```
I don't think it is supposed to work that way, but I may be mistaken.

I run this code inside Rails 5 app with `rails-i18n` gem.

**Comment by kbredemeier** [comment]
Same issue here. I pinned down the problem to `lib/dry/validation/schema/class_interface.rb:134`:
```ruby
def self.messages
  default = default_messages

  if config.messages_file && config.namespace
    default.merge(config.messages_file).namespaced(config.namespace)
  elsif config.messages_file
    default.merge(config.messages_file)
  elsif config.namespace
    default.namespaced(config.namespace)
  else
    default
  end
end

```
The `default_messages` is a `Dry::Validation::Messages::I18n` and has the correct `default_locale`.
When in the first `if` block `#namespace` gets called, it instantiates a `Dry::Validation::Messages::Namespaced` and passes the `Messages:I18n` message as the 2nd argument to the constructor. The `Messages::Namespaced` message inherits it's `default_local` from `Messages::Abstract` wich just returns `:en`.

A possible fix could be to override the '#default_locale' method on the `Messages::Namespaced` class to delegate to it's `@message`.

**Comment by solnic** [comment]
This is fixed in 0.10.6
